test_tools/stdin2rllib/main.cc

```diff
@@ -185,26 +190,29 @@ void parse_and_send_ca_event(json::value obj, r::live_model& rl, r::api_status&
 	}
 }
 
-void parse_and_send_cb_event(json::value obj, r::live_model& rl, r::api_status& status, bool debug) {
+void parse_and_send_cb_event(const rj::Document& obj, r::live_model& rl, r::api_status& status, bool debug) {
 	//parse event id
-	auto evt_id = U("EventId");
-	bool has_event_id = obj.has_field(evt_id);
+	auto evt_id = "EventId";
+	bool has_event_id = obj.HasMember(evt_id);
 	if (!has_event_id) {
 		std::cout << "missing 'EventId' field" << std::endl;
 		return;
 	}
-	std::string event_id = to_utf8string(obj.at(evt_id).as_string());
+	std::string event_id = obj[evt_id].GetString();
```

Comment:
GetString() returns a pointer. Do we need GetStringLength() for Unicode?
https://rapidjson.org/md_doc_tutorial.html#QueryString

Can we avoid the string copy, and pass the pointer to choose_rank() below?